### ISSUE TITLE
Redirect stdout from shell started from editor_scripts to editor console

### ISSUE THREAD

**Issue by Jerakin** [defold/editor2-issues#2781]
Right now you have to redirect it inside of the script to a log file all the time.

**Comment by Jerakin** [comment]
The more I try to use it the more I would like this. It is so hard to get any info when stuff goes wrong. 
I would like to do something as easy as
```
action = "shell", 
command = {"echo", "$PATH"}
```
But because it doesn't redirect the stdout I have to try to redirect it myself
```
action = "shell", 
command = {"echo", "$PATH", ">>", "./log.txt"}
```
Unfortunatelly this also silently fails, and I have to way to understand why.

**Comment by Jerakin** [comment]
Until this is fixed you can add this at the top of you python script
```
import sys
import traceback
def log_exception(_, _, tb):
    with open("python_log.txt", "a") as fp:
        traceback.print_tb(tb, file=fp)

sys.stdout = open("python_log.txt", 'w')
sys.excepthook = log_exception
```
It will print all your prints as well as any uncaught exceptions to a log file in the project root.

**Comment by vlaaad** [comment]
Fixed, check for updates!

**Comment by Jerakin** [comment]
First of this is super useful so thanks!

Right now it looks like the logging is collected/redirected when the editor script is finished. Would it be possible to "stream" it as it is printed?

**Comment by vlaaad** [comment]
uhh what, I thought I implemented streaming...

**Comment by vlaaad** [comment]
I tried running this script:
```
#!/usr/bin/env bash

echo "hello"
sleep 1
echo "world"
sleep 1
echo "good bye"
sleep 1
echo "world"
```
...and it displayed every new line of output with 1 second delay, as expected. Can you share a repro? What's your editor sha?

**Comment by Jerakin** [comment]
It's from python from a .editor_script
```
local script = 'import time\nfor i in range(4):\n    print("print " + str(i+1))\n    time.sleep(1)'
----

{
    action = "shell",
    command = {"python", "-c", script}
}
```

Works as expected in a terminal window

```

python -c 'import time
for i in range(10):
    print("print " + str(i))
    time.sleep(1)'

```

Have the most recent version
![Screenshot 2019-10-23 at 10 42 40](https://user-images.githubusercontent.com/5273682/67374701-daaf0400-f581-11e9-9384-eedc739bcc07.png)

**Comment by vlaaad** [comment]
TLDR: Use `{"python", "-u", "-c", script}`

Investigated what happens: it seems python checks if it's running in terminal or not and uses different buffering foru output/error streams. You can observe same behavior when piping terminal invocation to `cat`:
```sh

python -c 'import time
for i in range(10):
    print("print " + str(i))
    time.sleep(1)' | cat
```

There is a `-u` flag you can supply to python to make it's output/error streams unbuffered, using this flag it will make editor able to read it's output immediately.

**Comment by Jerakin** [comment]
Oh cool! Thanks a lot
